tests/test_generate_ics.py

```
@@ -56,27 +56,53 @@ def temp_cwd():
-            str(context.exception), "Invalid years value: 'abc'. Expected YYYY or YYYY-YYYY"
+            str(context.exception),
+            "Invalid years value: 'abc'. Expected YYYY, YYYY-YYYY, +N, or -N",
         )
 
+    @patch("holidays.generate_ics.datetime", MockDatetime)
+    def test_parse_years_offset(self):
+        self.assertEqual(IcsGenerator.parse_years("+5"), (2025, 2030))
+        self.assertEqual(IcsGenerator.parse_years("-4"), (2021, 2025))
+
+    def test_parse_years_offset_invalid(self):
+        with self.assertRaises(ArgumentTypeError) as context:
```

Ruff (0.15.17)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)